### PR TITLE
chore: add proper pluralization

### PR DESCRIPTION
**Pull request by pacMakaveli**
Hey @mattbrictson 

I'm trying to write some tests for this update but I'm having issues getting the output to give me what I want.

```    def test_pluralization_in_interactive_list
      stdout, stderr, status = Dir.chdir(File.expand_path("../fixtures", __dir__)) do
        VCR.use_cassette("changelog_requests", :record => :new_episodes) do
          unchanged_lockfile = File.read("Gemfile.lock")

          BundlerCommands.expects(:parse_outdated).returns({ "rails" => "" })
          BundlerCommands.expects(:read_updated_lockfile).returns(unchanged_lockfile)
          mock_vulnerable_gems(['rails'])

          capture_io_and_exit_status do
            CLI.new.run(argv: [])
          end
        end
      end
      require 'pry'; binding.pry

      assert_equal(<<~EXPECTED_STDERR, stderr)
        Resolving latest gem versions...
        Checking for security vulnerabilities...
        Finding changelogs.
      EXPECTED_STDERR

      assert_match(/1 gem can be updated/, stdout)
      assert_nil(status)
    end
```

I tried this test to no avail. FYI, the ` BundlerCommands.expects(:parse_outdated).returns({ "rails" => "" })` was initially  ` "sqlite3" => "2.0.3" `.
I'm assuming it's something to do with `stdin_data = " j j \n"` or maybe my setup is not right.

Would you be able to help please?

### CODE REVIEW

**Comment by mattbrictson** [comment]
Hi @pacMakaveli, thanks for looking into this!

Right now `cli_test.rb` is pretty close to an integration test. It operates on a real `Gemfile` that is located in `test/fixtures`. The updatable gems are determined by comparing `fixtures/Gemfile.lock` with `fixtures/Gemfile.lock.updated` (the `.updated` fixture simulates what happens by running `bundle lock --update`).

The `BundlerCommands.expects(:parse_outdated)` mock is actually for a secondary feature of determining "withheld" gems, so changing that mock isn't going to affect the "gems can be updated" text.

I think the easiest way forward is:

Create a new copy of `fixtures/Gemfile.lock.updated`, maybe called `fixtures/Gemfile.lock.updated-one`, and manually edit it so that only one gem version is different from `fixtures/Gemfile.lock`. Then change the `BundlerCommands.expects(:read_updated_lockfile)` mock to return the contents of that new fixture.

Does that sound reasonable?

**Comment by mattbrictson** [comment]
@pacMakaveli I recently modified `cli_test.rb` on the main branch to make it closer to a pure unit test. To write your test, you should be able to use the `stub_report` helper, which makes setting up different scenarios easier now. See #41.

**Comment by mattbrictson** [comment]
Something like this:

```ruby
def test_uses_correct_grammar_when_only_one_gem_can_be_updated
  report = stub_report(
    updatable_gems: {
      "sqlite3" => build(:outdated_gem, name: "sqlite3", changelog_uri: nil)
    }
  )
  report.expects(:scan_for_vulnerabilities!)

  stdout, = capture_io_and_exit_status(stdin_data: "\n") do
    CLI.new.run(argv: [])
  end

  assert_includes stdout, "1 gem can be updated"
end
```

**Comment by pacMakaveli** [comment]
Thanks @mattbrictson . I've not had a chance to look at it, but hopefully I can get to it by the end of today, specially with you doing most of my homework. 👯

**Comment by pacMakaveli** [comment]
Also, I had to add this in `test_helper`
```

require 'active_support/core_ext/module/delegation'
require 'active_support/inflector'
```

Otherwise I get this:
```
rake
/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.2.0/lib/active_support/delegation.rb:47:in `generate': uninitialized constant #<Class:ActiveSupport::Delegation>::Inflector (NameError)

          unless Inflector.safe_constantize(to.name).equal?(to)
                 ^^^^^^^^^
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.2.0/lib/active_support/core_ext/module/delegation.rb:161:in `delegate'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/factory_bot-6.4.5/lib/factory_bot/definition_hierarchy.rb:3:in `<class:DefinitionHierarchy>'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/factory_bot-6.4.5/lib/factory_bot/definition_hierarchy.rb:2:in `<module:FactoryBot>'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/factory_bot-6.4.5/lib/factory_bot/definition_hierarchy.rb:1:in `<top (required)>'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:97:in `require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:97:in `require'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.17/lib/zeitwerk/kernel.rb:34:in `require'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/factory_bot-6.4.5/lib/factory_bot.rb:8:in `<top (required)>'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:160:in `require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:160:in `rescue in require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:40:in `require'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.17/lib/zeitwerk/kernel.rb:34:in `require'
	from /Users/vlad/Projects/bundle_update_interactive/test/support/factory_bot.rb:3:in `<top (required)>'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:149:in `require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:149:in `require'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.17/lib/zeitwerk/kernel.rb:34:in `require'
	from /Users/vlad/Projects/bundle_update_interactive/test/test_helper.rb:9:in `block in <top (required)>'
	from /Users/vlad/Projects/bundle_update_interactive/test/test_helper.rb:9:in `each'
	from /Users/vlad/Projects/bundle_update_interactive/test/test_helper.rb:9:in `<top (required)>'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from /Users/vlad/Projects/bundle_update_interactive/test/bundle_update_interactive/bundler_commands_test.rb:3:in `<top (required)>'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:21:in `block in <main>'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:6:in `select'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:6:in `<main>'
<internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:149:in `require': cannot load such file -- factory_bot (LoadError)
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:149:in `require'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.17/lib/zeitwerk/kernel.rb:34:in `require'
	from /Users/vlad/Projects/bundle_update_interactive/test/support/factory_bot.rb:3:in `<top (required)>'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:149:in `require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:149:in `require'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.17/lib/zeitwerk/kernel.rb:34:in `require'
	from /Users/vlad/Projects/bundle_update_interactive/test/test_helper.rb:9:in `block in <top (required)>'
	from /Users/vlad/Projects/bundle_update_interactive/test/test_helper.rb:9:in `each'
	from /Users/vlad/Projects/bundle_update_interactive/test/test_helper.rb:9:in `<top (required)>'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from /Users/vlad/Projects/bundle_update_interactive/test/bundle_update_interactive/bundler_commands_test.rb:3:in `<top (required)>'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from <internal:/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:21:in `block in <main>'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:6:in `select'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:6:in `<main>'
rake aborted!
Command failed with status (1)

Tasks: TOP => default => test
(See full trace by running task with --trace)
```

**Review by mattbrictson — CHANGES_REQUESTED** [review]
Thanks, this is looking good! Could you fix the RuboCop issues flagged by CI? You can also check them locally by running `bundle exec rubocop` and auto-fix many of them with `bundle exec rubocop -A`.

Also I have a couple specific requests, see PR review comments below.

File: lib/bundle_update_interactive/string_helper.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# lib/bundle_update_interactive/string_helper.rb
+module BundleUpdateInteractive
+  module StringHelper
+    def pluralize(count, singular, plural = nil)
+      plural ||= "#{singular}s"
+      "#{count} #{count == 1 ? singular : plural}"
+    end
+    module_function :pluralize
+  end
+end
```
[the remark this answers is not in the document]
Comment (pacMakaveli) [reply lib/bundle_update_interactive/string_helper.rb:12]: Sure thing. Maybe you have it installed globally, which is why my RubyMine didn't pick the need to auto-fix the issues?

```
bundle exec rubocop
bundler: failed to load command: rubocop (/Users/vlad/.asdf/installs/ruby/3.2.2/bin/rubocop)
/Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/rubygems_integration.rb:265:in `block in replace_bin_path': can't find executable rubocop for gem rubocop. rubocop is not currently included in the bundle, perhaps you meant to add it to your Gemfile? (Gem::Exception)
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/rubygems_integration.rb:293:in `block in replace_bin_path'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/bin/rubocop:25:in `<top (required)>'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/cli/exec.rb:58:in `load'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/cli/exec.rb:23:in `run'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/cli.rb:455:in `exec'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/cli.rb:35:in `dispatch'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/cli.rb:29:in `start'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/exe/bundle:28:in `block in <top (required)>'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.5.17/exe/bundle:20:in `<top (required)>'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/bin/bundle:25:in `load'
	from /Users/vlad/.asdf/installs/ruby/3.2.2/bin/bundle:25:in `<main>'
```

File: test/test_helper.rb [hunk 2]
```diff
@@ -4,6 +4,9 @@
 require "bundle_update_interactive"
 require "minitest/autorun"
 
+require 'active_support/core_ext/module/delegation'
+require 'active_support/inflector'
```
[the remark this answers is not in the document]
Comment (pacMakaveli) [reply test/test_helper.rb:8]: You are right! How odd.. `bundle exec rake` is working fine.

**Pull request merged**